### ISSUE TITLE
[Feature Request] MAL search prefix for id and user's list

### ISSUE THREAD

**Issue by scb261** [tachiyomiorg/tachiyomi#4240]
## Why/User Benefit/User Problem
Recently myanimelist tracking was rewritten, and id search was added https://github.com/tachiyomiorg/tachiyomi/commit/a49adbd09c606408827da252e6a5be9c4a19fe89 as workaround for bad results. It uses `my:` for id search, but `my:` was used to search in user's list in older versions according to https://github.com/tachiyomiorg/tachiyomi/issues/65#issuecomment-175080484 and it confuses users now. Looking at older code, it looks like it was parsing user's list and searching in it locally:
https://github.com/tachiyomiorg/tachiyomi/blob/722437a022bb9e0a033682bf4e6034ced5af1047/app/src/main/java/eu/kanade/tachiyomi/data/track/myanimelist/MyAnimeListApi.kt#L39-L45

## What/Requirements

- [x] Search by id should begin with `id:`, not `my:`.
- [x] Use `my:` to search in user's list. I didn't find search in the list with api, so probably just [get the list](https://myanimelist.net/apiconfig/references/api/v2#operation/users_user_id_mangalist_get) and search locally.

**Comment by arkon** [comment]
Good catch, definitely a miss on my part since I don't actually use MAL.

**Comment by Aedyrn** [comment]
As this is related to the issue I brought up I think it might be pertinent to post this here rather than open a new issue. I went through the about 500 manga I have on tachiyomi looking for a pattern for which ones track and which ones don't. Eventually I noticed that 99% of them were tagged as "ecchi" on MAL and the ones that weren't clearly were ecchi but had not had their tags updated. So I checked the API page and noticed the NSFW section had this there:

> **Not Safe For Work**
By default, some APIs don’t return nsfw content.

And manga details responses had an nsfw tag I can't see on their website.

>`"nsfw": "white",`

Considering everyone of the manga I couldn't find had some amount or frequently focused on fanservice, including relatively major ones like Yuragi-sou no Yuuna-san. So I assume this means some manga are tagged as NSFW, and this is making them get filtered.

Now assuming this is the case, I don't know what the correct answer is as really all of those I can't find certainly aren't SFW.

Something like https://myanimelist.net/manga/111770/Kawaikereba_Hentai_demo_Suki_ni_Natte_Kuremasu_ka is certainly NSFW just based off of the MAL cover alone.

Maybe something like a setting or using a `nsfw:` tag as a means of searching with nfsw content on and still having it by default off?
Then again I'm pretty sure iirc Anilist tracking provides all manga as a response, including 18+ ones, so maybe just having nfsw content always on would be better.
